fatiando/gravmag/transform.py

```diff
@@ -503,6 +507,105 @@ def derivz(x, y, data, shape, order=1, method='fft'):
     return deriv[padx: padx + nx, pady: pady + ny].ravel()
 
 
+def power_density_spectra(x, y, data, shape):
+    r"""
+    Calculates the Power Density Spectra of a 2D gridded potential field
+    through the FFT:
+
+    .. math::
+
+        \Phi_{\Delta T}(k_x, k_y) = | F\left{\Delta T \right}(k_x, k_y) |^2
+
+    .. note:: Requires gridded data.
+
+    .. note:: x, y, z and height should be in meters.
+
+    Parameters:
+
+    * x, y : 1D-arrays
+        The x and y coordinates of the grid points
+    * data : 1D-array
+        The potential field at the grid points
+    * shape : tuple = (nx, ny)
+        The shape of the grid
+
+    Returns:
+
+    * kx, ky : 2D-arrays
+        The wavenumbers of each Power Density Spectra point
+    * pds : 2D-array
+        The Power Density Spectra of the data
+    """
+    kx, ky = _fftfreqs(x, y, shape, shape)
+    pds = abs(numpy.fft.fft2(numpy.reshape(data, shape)))**2
+    return kx, ky, pds
+
+
+def radial_average(kx, ky, pds, max_radius=None, ring_width=None):
```

Comment:
First I thought to make it explicitly generic to any 2D array, but I noticed that the kx and ky arrays (as outputs of _fftfreq) has special properties:
- they always include the origin (kx, ky) = (0, 0)
- they are ordered in special way (first the positive and then the negatives)
- if the original array has even amount of points per axes then the kx and ky arrays are not symmetric to the origin, instead they have one extra point in the positive semiaxe.

So I decided to stick to only frequency domain (or wavenumber domain) 2D arrays only. Maybe we can generalize this function. In order to do that I think that we will have to overcome two possible problems:
- How the distances `\Delta k_x` and `\Delta k_y` are calculated.
- What would happen if the `x` and `y` arrays don't include the origin (that would lead to a nan in the `radial_pds` array)?